### ISSUE TITLE
mbed TLS crash (Floating point exception) on CentOS 3 x86_64

### ISSUE THREAD

**Issue by hrk141** [ARMmbed/mbedtls#484]
Hi, thank you for letting me use mbed TLS.

We found a bug on CentOS 3 x86_64.
I used mbed TLS version 1.3.16.
Following code affect floating point exception.
CentOS 4 to 7 and CentOS 3 ia32 can run correctly.

Do you still support CentOS 3?

```
#include "polarssl/bignum.h"

int main(int argc, char **argv){
    mpi a, q, b, r;
    mpi_init(&a);
    mpi_init(&q);
    mpi_init(&b);
    mpi_init(&r);
    mpi_lset(&a, argc>2 ? atoi(argv[1]) : 100);
    mpi_lset(&b, argc>2 ? atoi(argv[2]) : 7);
    if(mpi_div_mpi(&q, &r, &a, &b) != 0){
        return -1;
    }
    printf("%zu(a) = %zu(q) * %zu(b) + %zu(r)\n", a.p[0], q.p[0], b.p[0], r.p[0]);
    return 0;
}
```

```
$ cat /etc/redhat-release
CentOS release 3.9 (Final)

$ rpm -q gcc glibc | uniq
gcc-3.2.3-60
glibc-2.3.2-95.50

$ gcc test_mbedtls_div.c -I ../mbedtls-1.3.16/include/ ../mbedtls-1.3.16/library/libmbedtls.a

(Expect)
$ ./a.out
100(a) = 14(q) * 7(b) + 2(r)

(Actual)
$ ./a.out
Floating point exception
```

**Comment by ciarmcom** [comment]
ARM Internal Ref: IOTSSL-777

**Comment by sbutcher-arm** [comment]
Hi,

To answer your question, we support CentOS as much as we do any other contemporary Linux distribution. There should be no dependencies that prevent us from executing on Red Hat or CentOS.

However - you're running a pretty old version of CentOS, and equally old versions of gcc and glibc - none of which look like they continue to be supported, and none of which we have tested with.

I've tried reproducing your fault on my current personal Linux development platform - Ubuntu 16.04, gcc 5.3.1, and unsurprisingly I see no faults, and as you say, you don't see this fault on more recent versions of CentOS.

If you were reporting an issue in a recent version of CentOS, then we might consider installing that version of CentOS just to reproduce the fault, but my concern is you're using a version that doesn't seem to be supported any longer, and the issue is likely in the platform. CentOS 3 was released very early in AMD64 adoption, and I imagine there's a bug in the platform in that area.

If someone were to provide a pull request to fix the issue we would consider merging it, or a stack trace we would consider investigating it - but on the basis of this report there's not much for us to go on.

Is it possible for you to upgrade to a more recent version of CentOS?
